exec-service-wasmer/src/wasmer_metering.rs

```diff
@@ -28,18 +28,24 @@ impl MeteringGlobalIndexes {
 }
 
 #[derive(Debug)]
-pub struct Metering {
+pub(crate) struct Metering {
     points_limit: u64,
-    opcode_cost: Rc<OpcodeCost>,
-    global_indexes: Rc<RefCell<Option<MeteringGlobalIndexes>>>,
+    opcode_cost: Arc<OpcodeCost>,
+    breakpoints_middleware: Arc<Breakpoints>,
+    global_indexes: Mutex<Option<MeteringGlobalIndexes>>,
 }
 
 impl Metering {
-    pub fn new(points_limit: u64, opcode_cost: Rc<OpcodeCost>) -> Self {
+    pub(crate) fn new(
+        points_limit: u64,
+        opcode_cost: Arc<OpcodeCost>,
```

Comment:
Do we need the Arc here? Might hurt performance

Reply:
Keeping an eye out for bottlenecks. Meanwhile we can keep it.